test/distributions/test_distributions.py

```diff
@@ -103,694 +99,697 @@ def is_all_nan(tensor):
+                'cov_factor': torch.randn(5, 2, 1, requires_grad=True),
+                'cov_diag': torch.tensor([2.0, 0.25], requires_grad=True),
+            },
+            {
+                'loc': torch.randn(4, 3, requires_grad=True),
+                'cov_factor': torch.randn(3, 2, requires_grad=True),
+                'cov_diag': torch.tensor([5.0, 1.5, 3.], requires_grad=True),
+            }
+        ]),
+        Example(MultivariateNormal, [
+            {
+                'loc': torch.randn(5, 2, requires_grad=True),
+                'covariance_matrix': torch.tensor([[2.0, 0.3], [0.3, 0.25]], requires_grad=True),
+            },
+            {
+                'loc': torch.randn(2, 3, requires_grad=True),
+                'precision_matrix': torch.tensor([[2.0, 0.1, 0.0],
+                                                  [0.1, 0.25, 0.0],
+                                                  [0.0, 0.0, 0.3]], requires_grad=True),
+            },
+            {
+                'loc': torch.randn(5, 3, 2, requires_grad=True),
+                'scale_tril': torch.tensor([[[2.0, 0.0], [-0.5, 0.25]],
+                                            [[2.0, 0.0], [0.3, 0.25]],
+                                            [[5.0, 0.0], [-0.5, 1.5]]], requires_grad=True),
+            },
+            {
+                'loc': torch.tensor([1.0, -1.0]),
+                'covariance_matrix': torch.tensor([[5.0, -0.5], [-0.5, 1.5]]),
+            },
+        ]),
+        Example(Normal, [
+            {
+                'loc': torch.randn(5, 5, requires_grad=True),
+                'scale': torch.randn(5, 5).abs().requires_grad_(),
+            },
+            {
+                'loc': torch.randn(1, requires_grad=True),
+                'scale': torch.randn(1).abs().requires_grad_(),
+            },
+            {
+                'loc': torch.tensor([1.0, 0.0], requires_grad=True),
+                'scale': torch.tensor([1e-5, 1e-5], requires_grad=True),
+            },
+        ]),
+        Example(OneHotCategorical, [
+            {'probs': torch.tensor([[0.1, 0.2, 0.3], [0.5, 0.3, 0.2]], requires_grad=True)},
+            {'probs': torch.tensor([[1.0, 0.0], [0.0, 1.0]], requires_grad=True)},
+            {'logits': torch.tensor([[0.0, 0.0], [0.0, 0.0]], requires_grad=True)},
+        ]),
+        Example(OneHotCategoricalStraightThrough, [
+            {'probs': torch.tensor([[0.1, 0.2, 0.3], [0.5, 0.3, 0.2]], requires_grad=True)},
+            {'probs': torch.tensor([[1.0, 0.0], [0.0, 1.0]], requires_grad=True)},
+            {'logits': torch.tensor([[0.0, 0.0], [0.0, 0.0]], requires_grad=True)},
+        ]),
+        Example(Pareto, [
+            {
+                'scale': 1.0,
+                'alpha': 1.0
+            },
+            {
+                'scale': (torch.randn(5, 5).abs() + 0.1).requires_grad_(),
+                'alpha': (torch.randn(5, 5).abs() + 0.1).requires_grad_()
```

Comment:
I've added these `+ 0.1`'s here to fix the `macos-12-py3-arm64 / test` failure. Earlier, I added some prints to the failing `test_cdf_icdf_inverse` test and looked at the printouts from the CI log. I found out that the samples generated by the Pareto distribution sometimes contain infinity if the values in `alpha` are too close to 0, and then after those infinities go through `cdf` and `icdf`, they turn into `nan`'s, causing comparisons to fail. I copied the exact `scale` and `alpha` values from the failing test into the following snippet:

```python
import torch

for seed in range(100_000):
    torch.manual_seed(seed)
    print(seed)

    param = {
        'scale': torch.tensor([
            [1.7549, 1.1252, 1.1057, 1.2407, 0.8933],
            [0.7661, 0.9723, 0.2869, 1.1382, 0.1948],
            [0.3116, 0.5687, 0.0172, 0.2296, 0.1637],
            [0.6526, 0.8761, 0.0142, 2.0804, 1.3138],
            [0.8522, 2.6757, 0.9873, 0.6828, 1.1830]],
            dtype=torch.double, requires_grad=True),
        'alpha': torch.tensor([
            [0.3219, 2.0156, 0.5705, 1.0555, 0.0485],
            [0.1138, 0.1064, 0.4582, 0.3166, 0.0073],
            [0.3978, 0.3402, 0.4575, 2.7605, 0.3148],
            [0.0442, 1.2770, 1.3061, 1.4474, 0.6836],
            [0.1606, 1.1500, 0.2150, 0.2591, 0.1284]],
            dtype=torch.double, requires_grad=True)
    }

    dist = torch.distributions.Pareto(**param)
    samples = dist.sample(sample_shape=(20,))

    assert not samples.isinf().any()
```

The assert throws an error for me once it gets to `seed=6`, so there is some luck involved--the test evidently used to pass in CI just because we were getting lucky with the random seeds.

If I add `0.1` to `alpha` in the above script, it does not raise an error for any of the 100,000 seeds that the script tries